memcopy.h

```diff
@@ -29,9 +29,98 @@ static inline uint32_t load_short(const unsigned char *in, unsigned bits) {
 #endif
 }
 
+static inline unsigned char *copy_2_bytes(unsigned char *out, unsigned char *from) {
+    /* Copy a few bytes to make sure the loop below has a multiple of SZ bytes to be copied. */
+    uint16_t chunk;
+    unsigned sz = sizeof(chunk);
+    MEMCPY(&chunk, from, sz);
+    MEMCPY(out, &chunk, sz);
+    return out + sz;
+}
+
+static inline unsigned char *copy_4_bytes(unsigned char *out, unsigned char *from) {
+    /* Copy a few bytes to make sure the loop below has a multiple of SZ bytes to be copied. */
+    uint32_t chunk;
+    unsigned sz = sizeof(chunk);
+    MEMCPY(&chunk, from, sz);
+    MEMCPY(out, &chunk, sz);
+    return out + sz;
+}
+
+/* Copy byte by byte LEN bytes from FROM into OUT. Return OUT + LEN. */
+static inline unsigned char *memcopy_byte_by_byte(unsigned char *out, unsigned char *from, unsigned len) {
+    switch (len) {
+    default:
+        Assert(len < 8, "memcopy_byte_by_byte should be called with less than 8 bytes");
+    case 7:
+        *out++ = *from++;
```

Comment:
This might cause misalignment on architectures where unaligned read/writes are not allowed. It depends on how wise the compiler is but only in best case scenario memcopy reverts to byte-by-byte loop. Currently there is one supported arch (out of 4) that has strict alignment rules.